tests/upload_files.rs

```diff
@@ -72,14 +72,96 @@ fn uploading_files_is_prevented(server: TestServer) -> Result<(), Error> {
         .error_for_status()
         .is_err());
 
-    // After uploading, check whether the uploaded file is now getting listed.
+    // After uploading, check whether the uploaded file is NOT getting listed.
     let body = reqwest::blocking::get(server.url())?;
     let parsed = Document::from_read(body)?;
     assert!(!parsed.find(Text).any(|x| x.text() == test_file_name));
 
     Ok(())
 }
 
+/// This test runs the server with --allowed-upload-dir argument and 
+/// checks that file upload to a different directory is actually prevented.
+#[rstest]
+#[case(server_no_stderr(&["-u",  "--allowed-upload-dir", "someDir"]))]
+#[case(server_no_stderr(&["-u",  "--allowed-upload-dir", "someDir/some_sub_dir"]))]
+fn uploading_files_is_restricted(
+    #[case] server: TestServer
+) -> Result<(), Error> {
+    let test_file_name = "uploaded test file.txt";
+
+    // Then try to upload file to root directory (which is not the --allowed-upload-dir)
+    let form = multipart::Form::new();
+    let part = multipart::Part::text("this should not be uploaded")
+        .file_name(test_file_name)
+        .mime_str("text/plain")?;
+    let form = form.part("file_to_upload", part);
+
+    let client = Client::new();
+    // Ensure uploading fails and returns an error
+    assert!(client
+        .post(server.url().join("/upload?path=/")?)
+        .multipart(form)
+        .send()?
+        .error_for_status()
+        .is_err());
```

Comment:
I was just thinking, perhaps check that this is actually forbidden. Otherwise, this test might fail for a 404 and we wouldn't catch it. Maybe also check for the precise error message?

Reply:
Yes, that makes more sense.